packages/core/admin/ee/server/services/user.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const { pipe, castArray, map, toNumber } = require('lodash/fp');
+
+/** Checks if ee disabled users list needs to be updated
+ * @param {string} id
+ * @param {object} input
+ */
+const shouldUpdateEEDisabledUsersList = async (id, input) => {
+  const data = await strapi.db.query('strapi::ee-store').findOne({
+    where: { key: 'ee_disabled_users' },
+  });
+
+  if (!data?.value || data.value.length === 0) {
+    return;
+  }
+  const disabledUsers = JSON.parse(data.value);
+  const user = disabledUsers.find((user) => user.id === Number(id));
+  if (!user) {
+    return;
+  }
+
+  if (user.isActive !== input.isActive) {
+    const newDisabledUsersList = disabledUsers.filter((user) => user.id !== Number(id));
+    await strapi.db.query('strapi::ee-store').update({
+      where: { id: data.id },
+      data: { value: JSON.stringify(newDisabledUsersList) },
+    });
+  }
+};
+
+const castNumberArray = pipe(castArray, map(toNumber));
+
+const shouldRemoveFromEEDisabledUsersList = async (ids) => {
+  let idsToCheck;
+  if (typeof ids === 'object') {
+    idsToCheck = castNumberArray(ids);
+  } else {
+    idsToCheck = [Number(ids)];
+  }
+
+  const data = await strapi.db.query('strapi::ee-store').findOne({
+    where: { key: 'ee_disabled_users' },
+  });
+
+  if (!data?.value || data.value.length === 0) {
+    return;
+  }
+  const disabledUsers = JSON.parse(data.value);
+
+  const newDisabledUsersList = disabledUsers.filter((user) => !idsToCheck.includes(user.id));
+  await strapi.db.query('strapi::ee-store').update({
+    where: { id: data.id },
+    data: { value: JSON.stringify(newDisabledUsersList) },
+  });
+};
+
+const getCurrentActiveUserCount = async () => {
+  return strapi.db.query('admin::user').count({ where: { isActive: true } });
+};
+
+const getDisabledUserList = async () => {
+  return strapi.db.query('strapi::ee-store').findOne({
+    where: { key: 'ee_disabled_users' },
+  });
+};
```

Comment:
I think this function could expose directly the parsed JSON, so the controllers using this service do not need to parse it. wdyt?

Reply:
Also, maybe I am missing something about how users are stored, but why not doing:

`strapi.db.query('admin::user').find({where: { isActive: {"$eq": true}}})` 
Why is it necessary to store it in the ee-store?

Reply:
Kinda same thing as above, this is our reference list, not actual user list. We would use it to check the actual user list and see if users have made any changes to user records